src/store/aggregator/aggregator.sagas.ts

```diff
@@ -121,6 +128,58 @@ export function* fetchPausedTokens() {
   }
 }
 
+export function* fetchIncentive() {
+  try {
+
+    yield* put(aggregatorActions.setIncentivesLoading());
+
+    const rewardManager = yield* select(rewardManagerSelector);
+    const sendAmount = yield* select(sendAmountSelector);
+    const flowState = yield* select(flowStateSelector);
+
+    if (!rewardManager) {
+      throw new Error('Could not find RewardManager contract');
+    }
+
+    const startingTokenAddress = (yield* select(startingTokenAddressSelector)) ?? '';
+    const destinationTokenAddress = (yield* select(destinationTokenAddressSelector)) ?? '';
+
+    let incentive = BigNumber.from(0);
+    let receiveAmount = BigNumber.from(0);
+    let incentiveType = IncentiveType.none;
+
+    if(flowState === 'deposit' && startingTokenAddress) {
```

Comment:
I think you can unify these 2 `if`s because they are almost identical. You just need to extract `if` conditions to separate constants and use few ternary operators (for `incentiveType`, `incentiveMethod` and `receiveAmount`).

Reply:
tried it, and I think it's more confusing....